### PR TITLE
[DLP] Implemented de-identify with simple word list

### PR DESCRIPTION
**Pull request by Avani-Thakker-Crest**
## Description
[DLP] Implemented dlp_deidentify_simple_word_list with unit test cases. Java equivalent: https://cloud.google.com/dlp/docs/creating-custom-infotypes-dictionary#simple_word_list

Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [x] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [ ] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [x] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [x] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [x] Please **merge** this PR for me once it is approved.
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample

### CODE REVIEW

File: dlp/snippets/deid.py [hunk 1]
```diff
@@ -937,6 +937,80 @@ def deidentify_with_replace_infotype(project, item, info_types):
...
 
 
+# [START dlp_deidentify_simple_word_list]
+def deidentify_with_simple_word_list(
+    project,
+    input_str,
+    info_type,
+    word_list
+):
+    """Uses the Data Loss Prevention API to deidentify sensitive data in a
```
Comment (realjordanna) [inline dlp/snippets/deid.py:947]: de-identify (when in comments)

File: dlp/snippets/deid.py [hunk 2]
```diff
@@ -937,6 +937,80 @@ def deidentify_with_replace_infotype(project, item, info_types):
...
+          None; the response from the API is printed to the terminal.
+    """
+    # Import the client library
+    import google.cloud.dlp
+
+    # Instantiate a client.
+    dlp = google.cloud.dlp_v2.DlpServiceClient()
+
+    # Convert the project id into a full resource id.
+    parent = f"projects/{project}"
```
Comment (realjordanna) [inline dlp/snippets/deid.py:967]: move down to where you are using it

File: dlp/snippets/deid.py [hunk 3]
```diff
@@ -937,6 +937,80 @@ def deidentify_with_replace_infotype(project, item, info_types):
...
+        {
+            "info_type": {"name": info_type},
+            "dictionary": {"word_list": word_list}
+        }
+    ]
+
+    # Construct the configuration dictionary
+    inspect_config = {
+        "custom_info_types": custom_info_types,
+        "include_quote": True,
```
Comment (realjordanna) [inline dlp/snippets/deid.py:981]: this is not true in the other two code samples for this, why differ?

File: dlp/snippets/deid_test.py [hunk 4]
```diff
@@ -289,3 +289,35 @@ def test_deidentify_with_replace_infotype(capsys):
...
+    deid.deidentify_with_simple_word_list(
+        GCLOUD_PROJECT,
+        "My Room is rm-green",
+        "CUSTOM_ROOM_ID",
+        word_list
+    )
+
+    out, _ = capsys.readouterr()
+
+    assert "rm-green" not in out
```
Comment (realjordanna) [inline dlp/snippets/deid_test.py:305]: a easier to read test would just assert the entire de-identified string is what is expected, instead of having multiple asserts - that's also more inline how we normally test this

File: dlp/snippets/deid_test.py [hunk 5]
```diff
@@ -289,3 +289,35 @@ def test_deidentify_with_replace_infotype(capsys):
...
+    deid.deidentify_with_simple_word_list(
+        GCLOUD_PROJECT,
+        content_str,
+        "CUSTOM_ROOM_ID",
+        word_list
+    )
+
+    out, _ = capsys.readouterr()
+
+    assert "[CUSTOM_ROOM_ID]" not in out
```
Comment (realjordanna) [inline dlp/snippets/deid_test.py:321]: if the next one is true, this one can't be false, so no need for both

File: dlp/snippets/deid.py [hunk 6]
```diff
@@ -968,8 +1042,8 @@ def deidentify_with_replace_infotype(project, item, info_types):
         type=int,
         default=0,
         help="The maximum number of sensitive characters to mask in a match. "
-        "If omitted the request or set to 0, the API will mask any mathcing "
-        "characters.",
+             "If omitted the request or set to 0, the API will mask any mathcing "
```
Comment (realjordanna) [inline dlp/snippets/deid.py:1045]: can you make sure all of this added whitespace wasn't a corruption?

**Review by realjordanna — CHANGES_REQUESTED** [review]

File: dlp/snippets/deid.py (hunk 1 above)
Comment (Avani-Thakker-Crest) [reply dlp/snippets/deid.py]: Thank you for reviewing; updated in the next push. Please check

File: dlp/snippets/deid.py (hunk 2 above)
Comment (Avani-Thakker-Crest) [reply dlp/snippets/deid.py]: In other samples, "parent" is initialized at the start, though I have updated in the next push. Please review. Thank you!

File: dlp/snippets/deid.py (hunk 3 above)
Comment (Avani-Thakker-Crest) [reply dlp/snippets/deid.py]: Correct, updated in the next push, Thanks for reviewing.

File: dlp/snippets/deid_test.py (hunk 4 above)
Comment (Avani-Thakker-Crest) [reply dlp/snippets/deid_test.py]: Got it, updated in the next commit, Please review.

File: dlp/snippets/deid_test.py (hunk 5 above)
Comment (Avani-Thakker-Crest) [reply dlp/snippets/deid_test.py]: Updated!

File: dlp/snippets/deid.py (hunk 6 above)
Comment (Avani-Thakker-Crest) [reply dlp/snippets/deid.py]: thank you for pointing that out, Updated.

**Review by realjordanna — APPROVED** [review]

**Comment by leahecole** [comment]
It looks like maybe this is the unintended consequence of a merge instead of a rebase - @Avani-Thakker-Crest please remove files from the PR that weren't ones you edited. To do that from your fork, for each file, I believe the command will be
`git checkout origin/main -- <path to file you want to reset>`

You could also consider looking into removing the commits you didn't write from the PR ([this stackoverflow is decent](https://stackoverflow.com/questions/36168839/how-to-remove-commits-from-a-pull-request))

**Comment by leahecole** [comment]
temporarily adding the do not merge til we get it a little cleaned up :)

File: appengine/standard_python3/bundled-services/blobstore/django/noxfile_config.py [hunk 7]
```diff
@@ -30,8 +30,8 @@
     # to 'BUILD_SPECIFIC_GCLOUD_PROJECT' if you want to opt in using a
     # build specific Cloud project. You can also use your own string
     # to use your own Cloud project.
-    # "gcloud_project_env": "GOOGLE_CLOUD_PROJECT",
```
[the remark this answers is not in the document]
Comment (Avani-Thakker-Crest) [reply appengine/standard_python3/bundled-services/blobstore/django/noxfile_config.py:33]: I synced my fork branch since one of my PRs is merged, and that way other changes are also synced

**Comment by Avani-Thakker-Crest** [comment]
> temporarily adding the do not merge til we get it a little cleaned up :)

Hi @leahecole , thank you for sharing the details, those were very helpful :) . I tried rebasing the entire repo and that's the reason other files' commits were present. I have cleaned the things up now, Please check and merge. Thank you

File: dlp/snippets/deid.py [hunk 8]
```diff
@@ -937,6 +937,79 @@ def deidentify_with_replace_infotype(project, item, info_types):
...
+    input_str,
+    info_type,
+    word_list
+):
+    """Uses the Data Loss Prevention API to de-identify sensitive data in a
+      string by matching against custom word list.
+
+    Args:
+        project: The Google Cloud project id to use as a parent resource.
+        input_str: The string to deidentify (will be treated as text).
```
Comment (realjordanna) [inline dlp/snippets/deid.py:952]: is this consistent with the other samples ... seems like we normally call it content_to_inspect or content_to_deid?

File: dlp/snippets/deid.py [hunk 9]
```diff
@@ -937,6 +937,79 @@ def deidentify_with_replace_infotype(project, item, info_types):
...
+        info_type: A string representing info type to look for.
```
Comment (realjordanna) [inline dlp/snippets/deid.py:953]: why is this needed if this is about custom infotypes ?

**Review by realjordanna — CHANGES_REQUESTED** [review]

File: dlp/snippets/deid.py (hunk 8 above)
Comment (Avani-Thakker-Crest) [reply dlp/snippets/deid.py:952]: other deidentified sample uses "input_str" example: deidentify_with_mask, deidentify_with_redact.

File: dlp/snippets/deid.py (hunk 9 above)
Comment (Avani-Thakker-Crest) [reply dlp/snippets/deid.py:953]: A name to the custom infotype should be provided, If user is providing the "content string", "word_list", the name to the custom infotype should also be provided by user, as the name would be related to the other inputs they provide, is what I think. Yet let me know if this needs a change. Thank you

Comment (realjordanna) [reply dlp/snippets/deid.py:953]: then it's custom_info_type_name ... this comment doesn't make it clear this is a name to be used for the findings produced by the custom infotype ...

**Review by realjordanna — CHANGES_REQUESTED** [review]

File: dlp/snippets/deid.py [hunk 10]
```diff
@@ -937,6 +937,79 @@ def deidentify_with_replace_infotype(project, item, info_types):
 # [END dlp_deidentify_replace_infotype]
 
 
+# [START dlp_deidentify_simple_word_list]
+def deidentify_with_simple_word_list(
+    project,
+    input_str,
+    info_type,
+    word_list
```
Comment (dandhlee) [inline dlp/snippets/deid.py:945]: Add type hints.

File: dlp/snippets/deid.py [hunk 11]
```diff
@@ -937,6 +937,79 @@ def deidentify_with_replace_infotype(project, item, info_types):
...
+):
```
Comment (dandhlee) [inline dlp/snippets/deid.py:946]: Missing return annotation 
```suggestion
) -> None:
```

File: dlp/snippets/deid.py [hunk 12]
```diff
@@ -937,6 +937,79 @@ def deidentify_with_replace_infotype(project, item, info_types):
...
+
+    Args:
+        project: The Google Cloud project id to use as a parent resource.
+        input_str: The string to deidentify (will be treated as text).
+        info_type: A string representing info type to look for.
+            A full list of info type categories can be fetched from the API.
+        word_list: The list of strings to match against.
+
+    Returns:
+          None; the response from the API is printed to the terminal.
```
Comment (dandhlee) [inline dlp/snippets/deid.py:958]: If it returns `None` then the docstring is not required.

File: dlp/snippets/deid.py [hunk 13]
```diff
@@ -937,6 +937,79 @@ def deidentify_with_replace_infotype(project, item, info_types):
...
+    response = dlp.deidentify_content(
+        request={
+            "parent": parent,
+            "deidentify_config": deidentify_config,
+            "inspect_config": inspect_config,
+            "item": item,
+        }
+    )
+
+    print("De-identified Content: {}".format(response.item.value))
```
Comment (dandhlee) [inline dlp/snippets/deid.py:1007]: Use f-string instead of `format`, unless it is used with logging libraries.
```suggestion
    print(f"De-identified Content: {response.item.value}")
```

File: dlp/snippets/deid_test.py [hunk 14]
```diff
@@ -291,6 +291,35 @@ def test_deidentify_with_replace_infotype(capsys):
     assert "My favorite site is [URL]" in out
 
 
+def test_deidentify_with_simple_word_list(capsys):
+    word_list = ["RM-GREEN", "RM-YELLOW", "RM_ORANGE"]
+    deid.deidentify_with_simple_word_list(
+        GCLOUD_PROJECT,
+        "My Room is rm-green",
+        "CUSTOM_ROOM_ID",
+        word_list
```
Comment (dandhlee) [inline dlp/snippets/deid_test.py:300]: If you're using a vertical list of items, add a comma on the last item as well. This is not necessary if they're laid out horizontally, for example: `deid.deidentify_with_simple_word_list(var1, var2, var3)`.
```suggestion
        word_list,
```

File: dlp/snippets/deid_test.py [hunk 15]
```diff
@@ -291,6 +291,35 @@ def test_deidentify_with_replace_infotype(capsys):
...
+
+
+def test_deidentify_with_simple_word_list_ignores_insensitive_data(capsys):
+    content_str = "My Room is rm-red"
+    word_list = ["RM-GREEN", "RM-YELLOW", "RM_ORANGE"]
+    deid.deidentify_with_simple_word_list(
+        GCLOUD_PROJECT,
+        content_str,
+        "CUSTOM_ROOM_ID",
+        word_list
```
Comment (dandhlee) [inline dlp/snippets/deid_test.py:315]: Ditto.
```suggestion
        word_list,
```

**Review by dandhlee — CHANGES_REQUESTED** [review]
Reviewing for Python style.

File: dlp/snippets/deid.py (hunk 9 above)
Comment (Avani-Thakker-Crest) [reply dlp/snippets/deid.py:953]: yeah right, updated! Thank you for pointing that.

File: dlp/snippets/deid.py (hunk 11 above)
Comment (Avani-Thakker-Crest) [reply dlp/snippets/deid.py:946]: okay the same is not followed in other code samples.

File: dlp/snippets/deid.py (hunk 12 above)
Comment (Avani-Thakker-Crest) [reply dlp/snippets/deid.py:958]: That's how it is written in other code samples, Please let me know if the code should be consistent with other code samples or I should follow the styling that you suggested.

File: dlp/snippets/deid.py (hunk 13 above)
Comment (Avani-Thakker-Crest) [reply dlp/snippets/deid.py:1007]: yes, updated that.

File: dlp/snippets/deid_test.py (hunk 14 above)
Comment (Avani-Thakker-Crest) [reply dlp/snippets/deid_test.py:300]: okay got it, thank you for your comment, Updated that.

File: dlp/snippets/deid_test.py [hunk 16]
```diff
@@ -291,6 +291,35 @@ def test_deidentify_with_replace_infotype(capsys):
...
+
+    assert "My Room is [CUSTOM_ROOM_ID]" in out
+
+
+def test_deidentify_with_simple_word_list_ignores_insensitive_data(capsys):
+    content_str = "My Room is rm-red"
+    word_list = ["RM-GREEN", "RM-YELLOW", "RM_ORANGE"]
+    deid.deidentify_with_simple_word_list(
+        GCLOUD_PROJECT,
+        content_str,
```
[the remark this answers is not in the document]
Comment (Avani-Thakker-Crest) [reply dlp/snippets/deid_test.py:313]: okay got it. I used variable content_str with that intent that it is used twice in the test case, a) where we call the function and b) during "assert". If we want exact same string in assert, I felt declaring as variable would be better. Though I have made changes as you suggested. Thank you for your comment :)

File: dlp/snippets/deid.py (hunk 10 above)
Comment (Avani-Thakker-Crest) [reply dlp/snippets/deid.py:945]: That's a very good suggestion, but with this change, it would be inconsistent with other existing code samples. Let me know how should  I move ahead. Thank you.

File: dlp/snippets/deid.py [hunk 17]
```diff
@@ -937,6 +937,79 @@ def deidentify_with_replace_infotype(project, item, info_types):
 # [END dlp_deidentify_replace_infotype]
 
 
+# [START dlp_deidentify_simple_word_list]
+def deidentify_with_simple_word_list(
+    project,
+    input_str,
+    custom_info_type_name,
+    word_list,
```
Comment (dandhlee) [inline dlp/snippets/deid.py:945]: Add type hints.

Comment (Avani-Thakker-Crest) [reply dlp/snippets/deid.py]: Added.

File: dlp/snippets/deid.py (hunk 12 above)
Comment (Avani-Thakker-Crest) [reply dlp/snippets/deid.py]: Updated.

File: dlp/snippets/deid.py (hunk 11 above)
Comment (Avani-Thakker-Crest) [reply dlp/snippets/deid.py]: ok, updated.

File: dlp/snippets/deid.py (hunk 10 above)
Comment (Avani-Thakker-Crest) [reply dlp/snippets/deid.py]: Updated, please check

File: dlp/snippets/deid.py [hunk 18]
```diff
@@ -937,6 +937,76 @@ def deidentify_with_replace_infotype(project, item, info_types):
 # [END dlp_deidentify_replace_infotype]
 
 
+# [START dlp_deidentify_simple_word_list]
+def deidentify_with_simple_word_list(
+    project: str,
+    input_str: str,
+    custom_info_type_name: str,
+    word_list: list,
```
Comment (dandhlee) [inline dlp/snippets/deid.py:945]: What is it a list of? list of String?

File: dlp/snippets/deid.py [hunk 19]
```diff
@@ -937,6 +937,76 @@ def deidentify_with_replace_infotype(project, item, info_types):
...
+) -> None:
+    """Uses the Data Loss Prevention API to de-identify sensitive data in a
+      string by matching against custom word list.
+
+    Args:
+        project: The Google Cloud project id to use as a parent resource.
+        input_str: The string to deidentify (will be treated as text).
+        custom_info_type_name:The name of the custom info type to use.
```
Comment (dandhlee) [inline dlp/snippets/deid.py:953]:
```suggestion
        custom_info_type_name: The name of the custom info type to use.
```

**Review by dandhlee — APPROVED** [review]
LGTM for Python style. Two minor comments.

File: dlp/snippets/deid.py (hunk 18 above)
Comment (dandhlee) [reply dlp/snippets/deid.py:945]: Be specific with the type of list (you need to import from `typing` to annotate types inside):
```
from typing import List
    word_list: List[str],
```

starting 3.8 you should be able to use built-ins rather than from the `Typing` library:
```
    word_list: list[str],
```

starting 3.9 we'd recommend being specific with the type of list:
```
from collections.abc import Sequence
    word_list: Sequence[str],
```

Comment (Avani-Thakker-Crest) [reply dlp/snippets/deid.py]: okay got it, Updated , please check. Thank you for all the suggestions

File: dlp/snippets/deid.py [hunk 20]
```diff
@@ -937,6 +937,76 @@ def deidentify_with_replace_infotype(project, item, info_types):
 # [END dlp_deidentify_replace_infotype]
 
 
+# [START dlp_deidentify_simple_word_list]
+def deidentify_with_simple_word_list(
+    project: str,
+    input_str: str,
+    custom_info_type_name: str,
+    word_list: list[str],
```
Comment (dandhlee) [inline dlp/snippets/deid.py:945]:
```suggestion
    word_list: List[str],
```

You'll also need to import List
`from typing import List`

I was wrong - `list[str]` can be used after 3.9 not 3.8 😅

Comment (Avani-Thakker-Crest) [reply dlp/snippets/deid.py:945]: ohh okay, got it, thank you for your comment.

Comment (Avani-Thakker-Crest) [reply dlp/snippets/deid.py]: updated please check

File: dlp/snippets/deid_test.py [hunk 21]
```diff
@@ -299,6 +299,32 @@ def test_deidentify_with_replace_infotype(capsys):
     assert "My favorite site is [URL]" in out
 
 
+def test_deidentify_with_simple_word_list(capsys):
+    deid.deidentify_with_simple_word_list(
+        GCLOUD_PROJECT,
+        "My Room is rm-green",
```
Comment (annadata) [inline dlp/snippets/deid_test.py:305]: Please use "Patient was seen in RM-YELLOW then transferred to rm green."

File: dlp/snippets/deid_test.py [hunk 22]
```diff
@@ -299,6 +299,32 @@ def test_deidentify_with_replace_infotype(capsys):
...
+        "CUSTOM_ROOM_ID",
+        ["RM-GREEN", "RM-YELLOW", "RM_ORANGE"],
+    )
+
+    out, _ = capsys.readouterr()
+
+    assert "My Room is [CUSTOM_ROOM_ID]" in out
```
Comment (annadata) [inline dlp/snippets/deid_test.py:312]: The output will be "Patient was seen in [CUSTOM_ROOM_ID] then transferred to [CUSTOM_ROOM_ID]"

File: dlp/snippets/deid_test.py [hunk 23]
```diff
@@ -299,6 +299,32 @@ def test_deidentify_with_replace_infotype(capsys):
     assert "My favorite site is [URL]" in out
 
 
+def test_deidentify_with_simple_word_list(capsys):
+    deid.deidentify_with_simple_word_list(
+        GCLOUD_PROJECT,
+        "My Room is rm-green",
+        "CUSTOM_ROOM_ID",
+        ["RM-GREEN", "RM-YELLOW", "RM_ORANGE"],
```
Comment (annadata) [inline dlp/snippets/deid_test.py:307]: Did you use RM_ORANGE intentional? It works fine.

**Review by annadata — APPROVED** [review]

File: dlp/snippets/deid_test.py (hunk 21 above)
Comment (Avani-Thakker-Crest) [reply dlp/snippets/deid_test.py]: updated

File: dlp/snippets/deid_test.py (hunk 22 above)
Comment (Avani-Thakker-Crest) [reply dlp/snippets/deid_test.py]: updated

File: dlp/snippets/deid_test.py (hunk 23 above)
Comment (Avani-Thakker-Crest) [reply dlp/snippets/deid_test.py:307]: did not get this comment.

Comment (annadata) [reply dlp/snippets/deid_test.py:307]: @Avani-Thakker-Crest The other entries use hyphen, Orange uses underscore.

Comment (Avani-Thakker-Crest) [reply dlp/snippets/deid_test.py:307]: ohh yeahhhh, let me update this one.

Comment (Avani-Thakker-Crest) [reply dlp/snippets/deid_test.py]: updated, thank you for noticing :)

**Review by realjordanna — APPROVED** [review]
